From 8c730adcc6206d6b8b335cd578c048e862d64288 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 15 Apr 2020 12:26:00 +0200
Subject: [PATCH] lib: image_sign_info helper functions in SPL

Do not build image_sign_info helper functions in SPL if not needed.

Fixes: b983cc2da0ba ("lib: rsa: decouple rsa from FIT image verification")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 Kconfig         |  2 +-
 common/Kconfig  | 11 +++++++++++
 common/Makefile |  2 +-
 3 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/Kconfig b/Kconfig
index 1b0b6999d8..6038e93afb 100644
--- a/Kconfig
+++ b/Kconfig
@@ -447,7 +447,7 @@ config SPL_FIT_SIGNATURE
 	select SPL_FIT
 	select SPL_RSA
 	select SPL_RSA_VERIFY
-	select IMAGE_SIGN_INFO
+	select SPL_IMAGE_SIGN_INFO
 
 config SPL_LOAD_FIT
 	bool "Enable SPL loading U-Boot as a FIT (basic fitImage features)"
diff --git a/common/Kconfig b/common/Kconfig
index 3072651082..75d9dfdc33 100644
--- a/common/Kconfig
+++ b/common/Kconfig
@@ -1053,3 +1053,14 @@ config IMAGE_SIGN_INFO
 	select SHA256
 	help
 	  Enable image_sign_info helper functions.
+
+if IMAGE_SIGN_INFO
+
+config SPL_IMAGE_SIGN_INFO
+	bool
+	select SHA1
+	select SHA256
+	help
+	  Enable image_sign_info helper functions in SPL.
+
+endif
diff --git a/common/Makefile b/common/Makefile
index 702f2396cf..ad030e05eb 100644
--- a/common/Makefile
+++ b/common/Makefile
@@ -112,7 +112,7 @@ obj-$(CONFIG_ANDROID_BOOT_IMAGE) += image-android.o image-android-dt.o
 obj-$(CONFIG_$(SPL_TPL_)OF_LIBFDT) += image-fdt.o
 obj-$(CONFIG_$(SPL_TPL_)FIT) += image-fit.o
 obj-$(CONFIG_$(SPL_)MULTI_DTB_FIT) += boot_fit.o common_fit.o
-obj-$(CONFIG_IMAGE_SIGN_INFO) += image-sig.o
+obj-$(CONFIG_$(SPL_TPL_)IMAGE_SIGN_INFO) += image-sig.o
 obj-$(CONFIG_$(SPL_TPL_)FIT_SIGNATURE) += image-fit-sig.o
 obj-$(CONFIG_$(SPL_TPL_)FIT_CIPHER) += image-cipher.o
 obj-$(CONFIG_IO_TRACE) += iotrace.o
-- 
2.20.1

